diff --git a/src/platform/egl-dri/video.c b/src/platform/egl-dri/video.c
index c0e46c5e..c8f9f016 100644
--- a/src/platform/egl-dri/video.c
+++ b/src/platform/egl-dri/video.c
@@ -85,6 +85,7 @@
             do { if (DEBUG) arcan_warning("%lld:%s:%d:%s(): " fmt "\n",\
 						arcan_timemillis(), "egl-dri:", __LINE__, __func__,##__VA_ARGS__); } while (0)
 
+#define verbose_print debug_print
 #ifndef verbose_print
 #define verbose_print
 #endif
